infinispan-spring-boot-starter-embedded/src/main/java/org/infinispan/spring/starter/embedded/InfinispanEmbeddedAutoConfiguration.java

```diff
@@ -76,14 +75,14 @@ public DefaultCacheManager defaultCacheManager() throws IOException {
          if (infinispanGlobalConfigurer != null) {
             globalConfigurationBuilder.read(infinispanGlobalConfigurer.getGlobalConfiguration());
          } else {
-            globalConfigurationBuilder.jmx().domain(DEFAULT_JMX_DOMAIN).enable();
             globalConfigurationBuilder.transport().clusterName(infinispanProperties.getClusterName());
+            globalConfigurationBuilder.jmx().enable();
          }
-         globalConfigurationBuilder.defaultCacheName("default");
          globalConfigurationCustomizers.forEach(customizer -> customizer.customize(globalConfigurationBuilder));
          configurationCustomizers.forEach(customizer -> customizer.customize(configurationBuilder));
 
-         manager = new DefaultCacheManager(globalConfigurationBuilder.build(), configurationBuilder.build(), false);
+         manager = new DefaultCacheManager(globalConfigurationBuilder.build(), false);
+         manager.defineConfiguration("default", configurationBuilder.build());
```

Comment:
why defining `default` configuration?

Reply:
I'm confused by this class :rofl: 
AFAIK, there are 3 ways to configure a cache!? you have the `configurers`, `configurationCustomizers` and `cacheConfigurations`

Reply:
the configurationCustomizers are for the default configuration. the default configuration is done here, but there is no cache default created. 
I changed the code not to use the deprecated code anymore but keep the compatibility.
We can decide to break it and say don't use anymore, but we need to deprecated code first and then delete it.

Reply:
the compatibility is already broken. 
if you do `CacheManager.getCache()` it will throw an exception: `ISPN000433: A default cache has been requested, but no cache has been set as default for this container`
Defining a cache with the name `default` doesn't fix anything. It is a no-op actually.

Reply:
what do you suggest for 11?
removing the default cache or leaving this stuff ?
I rather clean up and align to infinispan's behaviour today

Reply:
default cache isn't going anywhere in Infinispan 11+ 
It is up to you if you want to keep it or not... But for this PR you have to support it and, if you want to remove it in the future, just deprecate.

Reply:
it's done